com.hazelcast.jet.impl.connector.StreamEventJournalP_WmCoalescingTest.when_allPartitionsIdleAndThenRecover_then_wmOutput

```
Error Message
processor output in mode "snapshots disabled, inboxLimit=1" doesn't match expected:<Watermark{IDLE_MESSAGE}
Watermark{ts=02:00:00.010}> but was:<Watermark{IDLE_MESSAGE}
Watermark{ts=02:00:00.010}
Watermark{IDLE_MESSAGE}>
```

https://hazelcast-l337.ci.cloudbees.com/view/Jet/job/jet-jdk9/com.hazelcast.jet$hazelcast-jet-core/286/testReport/junit/com.hazelcast.jet.impl.connector/StreamEventJournalP_WmCoalescingTest/when_allPartitionsIdleAndThenRecover_then_wmOutput/

fail again:

https://github.com/hazelcast/hazelcast-jet/issues/1003

```
Error Message

processor output in mode "snapshots disabled, inboxLimit=1" doesn't match expected:<Watermark{IDLE_MESSAGE}
Watermark{ts=02:00:00.012}> but was:<Watermark{IDLE_MESSAGE}
Watermark{ts=02:00:00.012}
Watermark{IDLE_MESSAGE}>

Stacktrace

java.lang.AssertionError: 
processor output in mode "snapshots disabled, inboxLimit=1" doesn't match expected:<Watermark{IDLE_MESSAGE}
Watermark{ts=02:00:00.012}> but was:<Watermark{IDLE_MESSAGE}
Watermark{ts=02:00:00.012}
Watermark{IDLE_MESSAGE}>
	at com.hazelcast.jet.core.test.JetAssert.throwNotEqual(JetAssert.java:63)
	at com.hazelcast.jet.core.test.JetAssert.assertEquals(JetAssert.java:46)
	at com.hazelcast.jet.core.test.TestSupport.runTest(TestSupport.java:580)
	at com.hazelcast.jet.core.test.TestSupport.expectOutputs(TestSupport.java:324)
	at com.hazelcast.jet.core.test.TestSupport.expectOutput(TestSupport.java:307)
	at com.hazelcast.jet.impl.connector.StreamEventJournalP_WmCoalescingTest.when_allPartitionsIdleAndThenRecover_then_wmOutput(StreamEventJournalP_WmCoalescingTest.java:152)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:105)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:97)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/StreamEventJournalP_WmCoalescingTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/StreamEventJournalP_WmCoalescingTest.java
@@ -22,6 +22,8 @@ import com.hazelcast.jet.config.JetConfig;
 import com.hazelcast.jet.core.JetTestSupport;
 import com.hazelcast.jet.core.Processor;
 import com.hazelcast.jet.core.Watermark;
+import com.hazelcast.jet.core.test.TestOutbox;
+import com.hazelcast.jet.core.test.TestProcessorContext;
 import com.hazelcast.jet.core.test.TestSupport;
 import com.hazelcast.jet.function.DistributedSupplier;
 import com.hazelcast.map.impl.proxy.MapProxyImpl;
@@ -33,6 +35,7 @@ import org.junit.runner.RunWith;
 
 import java.util.HashSet;
 import java.util.List;
+import java.util.Queue;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.Future;
 import java.util.concurrent.locks.LockSupport;
@@ -48,6 +51,8 @@ import static com.hazelcast.spi.properties.GroupProperty.PARTITION_COUNT;
 import static java.util.Arrays.asList;
 import static java.util.Collections.singletonList;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.stream.Collectors.toList;
+import static org.junit.Assert.assertEquals;
 
 @RunWith(HazelcastParallelClassRunner.class)
 public class StreamEventJournalP_WmCoalescingTest extends JetTestSupport {
@@ -131,25 +136,30 @@ public class StreamEventJournalP_WmCoalescingTest extends JetTestSupport {
     @Test
     public void when_allPartitionsIdleAndThenRecover_then_wmOutput() throws Exception {
         // Insert to map in parallel to verifyProcessor.
+        CountDownLatch latch = new CountDownLatch(1);
         Thread updatingThread = new Thread(() -> uncheckRun(() -> {
             // We will start after a delay so that the source will first become idle and then recover.
-            Thread.sleep(4000);
-            for (int i = 0; i < 32; i++) {
+            latch.await();
+            for (;;) {
                 map.put(partitionKeys[0], 12);
-                Thread.sleep(250);
+                Thread.sleep(100);
             }
         }));
         updatingThread.start();
 
-        TestSupport.verifyProcessor(createSupplier(asList(0, 1), 1000))
-                   .disableProgressAssertion()
-                   .disableRunUntilCompleted(8000)
-                   .disableSnapshots()
-                   .outputChecker((e, a) -> {
-                       a.removeAll(singletonList(12));
-                       return a.equals(e);
-                   })
-                   .expectOutput(asList(IDLE_MESSAGE, wm(12)));
+        Processor processor = createSupplier(asList(0, 1), 2000).get();
+        TestOutbox outbox = new TestOutbox(1024);
+        Queue<Object> outbox0 = outbox.queue(0);
+        processor.init(outbox, new TestProcessorContext());
+
+        assertTrueEventually(() -> {
+            processor.complete();
+            // after we have the IDLE_MESSAGE, release the latch to let the other thread produce events
+            if (IDLE_MESSAGE.equals(outbox0.peek())) {
+                latch.countDown();
+            }
+            assertEquals(asList(IDLE_MESSAGE, wm(12), 12), outbox0.stream().distinct().collect(toList()));
+        });
 
         updatingThread.interrupt();
         updatingThread.join();

